src/coreclr/pal/src/thread/thread.cpp

```diff
@@ -1581,6 +1581,17 @@ GetThreadTimes(
     return (retval);
 }
 
+VOID
+PALAPI
+GetThreadDescription(
+    IN HANDLE hThread,
+    IN size_t length,
+    OUT char* name)
+{
+    CPalThread *pThread = InternalGetCurrentThread();
```

Comment:
Can this be safely freed?

Reply:
Is hThread intentionally unused? or should we use it, e.g.:

```c++
    CPalThread *pCurrentThread;
    CPalThread *pTargetThread;
    IPalObject *pobjThread = NULL;

    pCurrentThread = InternalGetCurrentThread();

    PAL_ERROR palError = InternalGetThreadDataFromHandle(
        pCurrentThread,
        hThread,
        &pTargetThread,
        &pobjThread
        );

    pthread_getname_np(pTargetThread->GetPThreadSelf(), name, length);
```
(callsite may or may not pass `hThread == currentPalThread`)

Reply:
Yeah that's my bad...this should use this handle to get the CPalThread object

Reply:
This is fixed with the latest commit